test/e2e/configurer/chain/commands.go

```diff
@@ -444,3 +448,26 @@ func (n *NodeConfig) TxGovVote(from string, propID int, option govv1.VoteOption,
 
 	n.LogActionF("successfully submitted vote %s to prop %d", option, propID)
 }
+
+// submitRefundableTxWithAssertion submits a refundable transaction,
+// and asserts that the tx fee is refunded
+func (n *NodeConfig) SubmitRefundableTxWithAssertion(
+	f func(),
+	shouldBeRefunded bool,
+) {
+	// balance before submitting the refundable tx
+	submitterBalanceBefore, err := n.QueryBalances(n.PublicAddress)
+	require.NoError(n.t, err)
+
+	// submit refundable tx
+	f()
```

Comment:
should we wait for a block to know if the tx was included in the block? and then check new balances

Reply:
Currently I leave this as responsibility for the function caller. Somestimes there is more sophisticated check, e.g., query the inclusion of a finality signature. Wdyt?